### PR TITLE
WIP: 147 general compare optimization

### LINKED ISSUE #147: Generalcompare Optimization
# Issue

## Description

Optimization of the general comparison expression using the annotated types.

## Sub-tasks

Atomized tasks that cannot be broken down further.

- [x] Implementing the shortcut
- [x] Making sure spec still passes

### PR DESCRIPTION
**Pull request by maanpalle**
# Pull Request

## Description

Added the shortcut to the general compare class.

Closes #147 

## Changes

1. Added shortcut
2. Moved some checks to annotation time

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

**Review by HectorPeeters — APPROVED** [review]
Looks good to me!

File: src/expressions/operators/compares/Compare.ts [hunk 1]
```diff
@@ -46,6 +50,21 @@ class Compare extends Expression {
 			case 'greaterThanOrEqualOp':
 			case 'greaterThanOp':
 				this._compare = 'generalCompare';
+				if (
+					firstType &&
+					secondType &&
+					firstType.type !== ValueType.ITEM &&
+					secondType.type !== ValueType.ITEM &&
+					firstType.type !== ValueType.ARRAY &&
+					secondType.type !== ValueType.ARRAY
```
Comment (yuanzexiong) [inline src/expressions/operators/compares/Compare.ts:59]: What's gonna happen to array btw?

Comment (maanpalle) [reply src/expressions/operators/compares/Compare.ts:59]: It just isn't going to react to the shortcut

Comment (maanpalle) [reply src/expressions/operators/compares/Compare.ts:59]: THis is done because array is anotated with just ARRAY not the types it contains

Comment (yuanzexiong) [reply src/expressions/operators/compares/Compare.ts:59]: Oh okay I see thanks

**Review by yuanzexiong — APPROVED** [review]

**Pull request merged**